services/client/services.go

```diff
@@ -542,6 +579,10 @@ func validateAdvertisement(sid services.ServiceID, id string) error {
 	return nil
 }
 
+func leaderCacheKey(sid services.ServiceID, opts services.ElectionOptions) string {
+	return fmt.Sprintf("%s/%s", sid.String(), opts.String())
```

Comment:
We don't want it to be unique though, we just want to make sure that if `client.LeaderService(sid, opts)` is called twice with the same `sid` and `opts` that we return the already created `LeaderService` instance